### ISSUE TITLE
Prefer Dependency Injection instead of Mocking

### ISSUE THREAD

**Issue by DavidAntaramian** [SparkPost/elixir-sparkpost#13]
@richleland As I was going through the requirements for change implementation for #12 I realized that the tests depend significantly on mocking the behaviour of HTTPotion. Moving forward, would you be open to preferring a dependency injection model over mocking? This is the style preferred for Elixir core components (see ["Mocks and explicit contracts" – José Valim on Plataformatec Blog](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/)).

My thoughts would be to do this once #12 is complete but prior to further development.

**Comment by richleland** [comment]
Makes sense to me @DavidAntaramian we can tackle this right after #12 as you said.
